x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/AuthorizationService.java

```diff
@@ -767,13 +767,19 @@ private void authorizeBulkItems(
                                 request.remoteAddress(),
                                 authzInfo
                             );
+                            final IndicesAccessControl.DeniedIndices deniedIndices;
+                            if (indicesAccessControl.isRestrictedIndex(resolvedIndex)) {
```

Comment:
If we let `AuthorizationService` have a reference to `restrictedIndicesAutomaton`, we call just use that to tell whether an index is restricted and don't have to call `isRestrictedIndex`.